lib/js/jsbigints.nim

```diff
@@ -0,0 +1,229 @@
+## Arbitrary precision integers.
+## * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsbigints is designed to be used with the JavaScript backend.".}
+
+type JsBigInt* = ref object of JsRoot ## Arbitrary precision integer for JavaScript target.
+
+func big*(integer: SomeInteger): JsBigInt {.importjs: "BigInt(#)".} =
+  ## Constructor for `JsBigInt`.
+  runnableExamples:
+    doAssert big(1234567890) == big"1234567890"
+
+func big*(integer: cstring): JsBigInt {.importjs: "BigInt(#)".} =
+  ## Constructor for `JsBigInt`.
+  runnableExamples:
+    doAssert big"-1" == big"1" - big"2"
+
+func toLocaleString*(this: JsBigInt): cstring {.importjs: "#.$1()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toLocaleString*(this: JsBigInt; locales: cstring): cstring {.importjs: "#.$1(#)".} =
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+  # TODO: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString#Using_options
+  runnableExamples:
+    doAssert big"2147483647".toLocaleString("EN".cstring) == "2,147,483,647".cstring
+
+func toLocaleString*(this: JsBigInt; locales: openArray[cstring]): cstring {.importjs: "#.$1(#)".} =
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+  ## When requesting a language that may not be supported, include a fallback language.
+  runnableExamples:
+    doAssert big"2147483647".toLocaleString(["EN".cstring, "ES".cstring]) == "2,147,483,647".cstring
+
+func toString*(this: JsBigInt; radix: int): cstring {.importjs: "#.$1(#)".} =
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+  runnableExamples:
+    doAssert big"2147483647".toString(2) == "1111111111111111111111111111111".cstring
+
+func toString*(this: JsBigInt): cstring {.importjs: "#.toString()".} # asserted on $
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func `$`*(this: JsBigInt): string =
+  ## Return a string representation of `JsBigInt`.
+  runnableExamples: doAssert $big"1024" == "1024"
+  $toString(this)
+
+func asIntN*(width: int; bigInteger: JsBigInt): int {.importjs: "BigInt.$1(#, #)".} =
```

Comment:
return type is wrong, should be JsBigInt; and also runnableExample doesn't illustrate the wrapping behavior.

Use this instead:

```nim
func asIntN*(bits: int; a: JsBigInt): JsBigInt {.importjs: "BigInt.$1(#, #)".} =
  ## wrap `a` to a signed `JsBigInt` of `bits` bits, ie between -2^(bits-1) and 2^(bits-1)-1.
  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asIntN
  runnableExamples:
    doAssert asIntN(4, 7.big) == 7.big
    doAssert asIntN(4, 8.big) == -8.big
```

* do similarly with `asUintN`

Reply:
No, this downcast to an integer, like an actual integer, it can only be `cint` or `int`.

Reply:
node -e 'console.log(BigInt.asIntN(66, 2n**64n))'
18446744073709551616n